pwiz_tools/Skyline/Util/UtilIO.cs

```diff
@@ -1058,6 +1058,38 @@ public static bool IsTempZipFolder(string path, out string zipFileName)
             }
             return true;
         }
+
+        /// <summary>
+        /// Returns true if a new file can be created in directoryPath.
+        /// </summary>
+        public static bool IsWritable(string directoryPath)
+        {
+            if (!Directory.Exists(directoryPath))
+                throw new DirectoryNotFoundException(directoryPath);
+
+            try
+            {
+                // generate random filenames until the path doesn't exist (technically has a race condition, but extremely unlikely)
+                string randomFilepath;
+                do
+                {
+                    randomFilepath = Path.Combine(directoryPath, Path.GetRandomFileName());
+                } while (File.Exists(randomFilepath));
+
+                // create the file with write permission
+                using (new FileStream(randomFilepath, FileMode.Create, FileAccess.ReadWrite))
+                {
+                }
+
+                // cleanup the file
+                File.Delete(randomFilepath);
+                return true;
+            }
+            catch (IOException)

```

Comment:
I think you also need to catch other exception types here such as "UnauthorizedAccessException" which is not an IOException.
Might as well change this to "catch (Exception)".

Reply:
Good catch, I didn't know unauthorized access on I/O wouldn't be derived from IOException!